Orange API management user story

tsegismont opened the pull request:
** If you are updating a guide, please submit your pull request to the main repository: https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc **

gsmet commented on _posts/2025-03-12-orange-telco-core-network-api-management-with-quarkus.adoc:
```diff
@@ -0,0 +1,110 @@
+
+== Why Quarkus? A Framework Evaluation
+
+The team at Orange embarked on a comparative study to test alternatives to Spring Boot in a well-defined API wrapper exposure context.
```
It's unclear why they wanted a Spring Boot alternative?

patrice-conil replied:
Because spring startup time can be a barrier to dynamic pod management

gsmet commented on _posts/2025-03-12-orange-telco-core-network-api-management-with-quarkus.adoc:
```diff
@@ -0,0 +1,110 @@
+
+Transitioning from Spring to Quarkus was not overly complex, especially for those familiar with JAX-RS / Jakarta REST.
+
+The https://quarkus.io/guides/dev-services[Dev Services] significantly enhanced productivity in dev and test modes, but required careful consideration for their CI/CD environments, where no container runtime is availabe.
```
```suggestion
The https://quarkus.io/guides/dev-services[Dev Services] significantly enhanced productivity in dev and test modes, but required careful consideration for their CI/CD environments, where no container runtime is available.
```

gsmet commented on _posts/2025-03-12-orange-telco-core-network-api-management-with-quarkus.adoc:
```diff
@@ -0,0 +1,110 @@
+Native compilation produced compact native executables, but the process of producing them was lengthy.
+The team reserved it for the final build stage when absolutely necessary.
+
+Some fine-tuning was required to prevent class pruning issues.
```
Probably worth explaining this for people not familiar with GraalVM.

tsegismont replied:
Could you suggest a link to the doc for this? I have a (good?) idea of what Patrice meant, but I must admit I don't feel comfortable explaining it.

tsegismont replied:
Thank you Guillaume!

gsmet commented on _posts/2025-03-12-orange-telco-core-network-api-management-with-quarkus.adoc:
```diff
@@ -0,0 +1,110 @@
+
+Having worked extensively with Reactor, the transition to Mutiny had a learning curve.
+
+While effective and more intuitive thanks to its navigable API, it was perceived as more verbose for the simple cases.
```
Is it something we should improve? Maybe we could collect comparative examples and see how we can improve on that?

patrice-conil replied:
IMHO the way Reactor implements map/flatmap is more natural for developers using functional programming. I also had trouble handling long-running requests that required using an http callback... I had to go back to a CompletableFuture to be able to inject the backend response afterward. But that's probably just a lack of skills. ;)

gsmet commented on _posts/2025-03-12-orange-telco-core-network-api-management-with-quarkus.adoc:
```diff
@@ -0,0 +1,110 @@
+
+== Architecture Overview
+
+Orange designed a microservices-based architecture to expose 5G APIs efficiently:
```
I would drop the colon. It's weird with titles.
```suggestion
Orange designed a microservices-based architecture to expose 5G APIs efficiently.
```

gsmet reviewed (commented):
Nice! I added a few suggestions/questions.

gsmet commented:
@tsegismont LGTM. I added a suggestion for the native executable pruning. Feel free to adapt it (or ask me for more if it doesn't work for you).

tsegismont commented:
@gsmet how about changing the date of publication to give some time to @patrice-conil for a final review? Let's say Tuesday?

patrice-conil commented:
> @gsmet how about changing the date of publication to give some time to @patrice-conil for a final review? Let's say Tuesday?

That's fine with me, you can publish.

tsegismont commented:
Thank you @patrice-conil !

gsmet commented:
@tsegismont I let you adjust the two dates and squash. And then we can merge and it's going to be automatically published on Tuesday.

I can post on social media too if you can come up with some short content I can push to the various platforms.

tsegismont commented:
@gsmet I've adjusted the two dates to Monday and squashed.

Regarding the short content for social media, I think we could use the text of the introduction:

```asciidoc
As a global telecommunications leader, https://www.orange.com[Orange] has always been at the forefront of innovation.
Along with Deutsche Telekom, Telefónica, and Vodafone, Orange co-founded the https://camaraproject.org[Camara Project], an initiative aimed at simplifying the consumption of 5G APIs for third-party application developers.

To achieve this goal, Orange needed a flexible and lightweight framework capable of handling constrained API exposure while ensuring compatibility with existing network core systems.

After a rigorous evaluation of multiple frameworks and toolkits — including Quarkus, Ktor, Micronaut, and Vert.x — Orange selected Quarkus as the ideal solution.

https://www.linkedin.com/in/patriceconil/[Patrice Conil], software craftsman at Orange, told us why.
```

tsegismont commented:
Shout-out to @geoand for helping with the writing, and thank you @patrice-conil for the feedback on Quarkus !

tsegismont commented:
@gsmet @geoand can one of you please merge the PR? I can't (not granted to do so). Thanks!
